drivers/sensor/mockup/battery/battery.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <kernel.h>
+#include <sensor.h>
+
+struct battery_data {
+	u8_t battery_lvl;
+};
+
+static struct device *device;
+static int battery_init(struct device *dev);
+static sensor_trigger_handler_t trigger_callback;
+
+static void trigger_fire(struct k_work *work)
+{
+	struct sensor_trigger trigger = {
+		.chan = SENSOR_CHAN_SOC,
+		.type = SENSOR_TRIG_DATA_READY
+	};
+
+	trigger_callback(device, &trigger);
+}
+
+K_WORK_DEFINE(timer_work, trigger_fire);
+
+static void timer_expire(struct k_timer *dummy)
+{
+	k_work_submit(&timer_work);
```

Comment:
You could have used a delayed_work here of a timer since you end up submitting a work anyway.